src/components/shared/Entity/prompts/steps/dataFlowReset/SelectMaterialization/Selector.tsx

```diff
@@ -1,28 +1,60 @@
-import { Autocomplete, Grid, TextField } from '@mui/material';
+import { Autocomplete, Grid, Skeleton, TextField } from '@mui/material';
 import { autoCompleteDefaults_Virtual } from 'components/shared/AutoComplete/DefaultProps';
-import { ReactNode, useState } from 'react';
+import { ReactNode, useEffect, useState } from 'react';
 import { useIntl } from 'react-intl';
-import { useBindingStore } from 'stores/Binding/Store';
+import { useLoopIndex } from 'context/LoopIndex/useLoopIndex';
+import { LiveSpecsExt_Related } from 'api/liveSpecsExt';
+import AlertBox from 'components/shared/AlertBox';
+import { usePreSavePromptStore } from '../../../store/usePreSavePromptStore';
 import { RelatedMaterializationSelectorProps } from './types';
 import SelectorOption from './SelectorOption';
 
 const getValue = (option: any) =>
     typeof option === 'string' ? option : option?.catalog_name;
 
-function Selector({ disabled, keys }: RelatedMaterializationSelectorProps) {
+function Selector({
+    disabled,
+    keys,
+    loading,
+}: RelatedMaterializationSelectorProps) {
     const intl = useIntl();
+    const stepIndex = useLoopIndex();
 
     const [inputValue, setInputValue] = useState('');
 
-    const [backfillDataFlowTarget, setBackfillDataFlowTarget] = useBindingStore(
-        (state) => [
-            state.backfillDataFlowTarget,
-            state.setBackfillDataFlowTarget,
-        ]
-    );
+    const [updateStep, updateContext] = usePreSavePromptStore((state) => [
+        state.updateStep,
+        state.updateContext,
+    ]);
+
+    const backfillTarget = usePreSavePromptStore((state) => {
+        state.context.backfillTarget;
+    });
+
+    useEffect(() => {
+        updateContext({
+            noMaterializations: Boolean(!loading && keys.length === 0),
+        });
+    }, [keys.length, loading, updateContext]);
+
+    if (loading) {
+        return <Skeleton />;
+    }
```

Comment:
Moved the loading state to inside the selector. This way the selector can also handle the "empty" state which needs to eventually allow steps to be skipped.